### PR TITLE
Improve migration guide

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [ ] Added tests for code changes _or_ test/build only changes
- [ ] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

My goal was to improve the migration guide dispersing the fog around the support of the retry feature an error handling.

## Approach

* applied code style on the links for new operations 7517927ee0261c152aa13f812fc7ae040cae95a3
* fixed incorrect links (8548ee067aeffce6f2b9ff186a3b3432295db9f6):
  * the link for `postChanges` has changed
  * the link for `postPartitionFind` has changed
  * `postMissingRevs` is excluded from the SDKs, I replaced it with `postRevsDiff`
* Added a warning about the error handling 4103af85a771cbf6a5235b0a7262809873029f5a
* Added a note about Replay adapters replacement 4103af85a771cbf6a5235b0a7262809873029f5a

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A build or packaging only changes

## Monitoring and Logging

- "No change"

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ remote HTTP API. For example, in the case of the document context manager, this
 to fetch and a `put_document` to save.
 1. In `cloudant-python-sdk` View, Search, and Query (aka `_find` endpoint) operation responses contain raw JSON
 content like using `raw_result=True` in `python-cloudant`.
+1. Replay adapters are replaced by the [automatic retries](https://github.
+   com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
+1. Error handling is not transferable from `python-cloudant` to `cloudant-python-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=python#error-handling) in our API docs.
+1. Configuring HTTP client is not transferable from `python-cloudant` to `python-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```
Comment (emlaver) [inline MIGRATION.md:23]: Maybe we should say something like:
```suggestion
1. Custom HTTP client configurations in `python-cloudant` are not transferable to `python-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -44,56 +48,55 @@ The table below contains a list of `python-cloudant` functions and the `cloudant
...
+|`db.get_view_result()`|[`postView`](https://cloud.ibm.com/apidocs/cloudant?code=python#postview)|
+|`db.list_design_documents()`|[`postDesignDocs`](https://cloud.ibm.com/apidocs/cloudant?code=python#postdesigndocs)|
+|`db.get_query_result()`|[`postFind`](https://cloud.ibm.com/apidocs/cloudant?code=python#postfind)|
+|`db.get_query_indexes()`|[`getIndexesInformation`](https://cloud.ibm.com/apidocs/cloudant?code=python#getindexesinformation)|
+|`db.create_query_index()`|[`postIndex`](https://cloud.ibm.com/apidocs/cloudant?code=python#postindex)|
+|`db.delete_query_index()`|[`deleteIndex`](https://cloud.ibm.com/apidocs/cloudant?code=python#deleteindex)|
+|`Document(db, '_local/docid').fetch()`|[`getLocalDocument`](https://cloud.ibm.com/apidocs/cloudant?code=python#getlocaldocument)|
+|`Document(db, '_local/docid').save()`|[`putLocalDocument`](https://cloud.ibm.com/apidocs/cloudant?code=python#putlocaldocument)|
+|`Document(db, '_local/docid').delete()`|[`deleteLocalDocument`](https://cloud.ibm.com/apidocs/cloudant?code=python#deletelocaldocument)|
+|`db.missing_revisions()/db.revisions_diff`|[`postRevsDiff`](https://cloud.ibm.com/apidocs/cloudant?code=python#postrevsdiff)|
```
Comment (emlaver) [inline MIGRATION.md:83]:
```suggestion
|`db.missing_revisions()/db.revisions_diff()`|[`postRevsDiff`](https://cloud.ibm.com/apidocs/cloudant?code=python#postrevsdiff)|
```

**Review by emlaver — APPROVED** [review]
Looks good! A couple of comments

File: MIGRATION.md (hunk 2 above)
Comment (mojito317) [reply MIGRATION.md]: Done in 697ad31

File: MIGRATION.md (hunk 1 above)
Comment (mojito317) [reply MIGRATION.md]: Done in 697ad31

**Review by eiri — APPROVED** [review]

File: MIGRATION.md [hunk 3]
```diff
@@ -17,6 +17,10 @@ remote HTTP API. For example, in the case of the document context manager, this
 to fetch and a `put_document` to save.
 1. In `cloudant-python-sdk` View, Search, and Query (aka `_find` endpoint) operation responses contain raw JSON
 content like using `raw_result=True` in `python-cloudant`.
+1. Replay adapters are replaced by the [automatic retries](https://github.
+   com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
+1. Error handling is not transferable from `python-cloudant` to `cloudant-python-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=python#error-handling) in our API docs.
+1. Custom HTTP client configurations in `python-cloudant` are not transferable to `python-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```
Comment (vmatyusGitHub) [inline MIGRATION.md:23]: What is `python-java-sdk`? Shouldn't it be `cloudant-python-sdk`?

Comment (mojito317) [reply MIGRATION.md:23]: woops, will fix it in #512

Comment (vmatyusGitHub) [reply MIGRATION.md:23]: Okay! Thanks!

**Pull request merged**